docs/guides/basic-rbac.mdx

```diff
@@ -3,267 +3,303 @@ title: Implement basic Role Based Access Control (RBAC) with metadata
 
-This guide assumes that you are using Next.js App Router. The concepts here can be adapted to Next.js Pages Router and Remix.
+This guide assumes that you're using Next.js App Router, but the concepts can be adapted to any React-based framework.
 
-## Configure the session token
+<Steps>
+  ### Configure the session token
 
-Clerk provides [user metadata](/docs/users/metadata#user-metadata), which is a tool that can be leveraged to build flexible custom logic into your application. Metadata can be used to store information, and in this case, it can be used to store a user's role.
+  Clerk provides [user metadata](/docs/users/metadata#user-metadata), which can be used to store information, and in this case, it can be used to store a user's role.
 
-`unsafeMetadata` can be read and updated in the browser, and because the user could modify this metadata it should be treated as unsafe and validated by your application before trusting it. `privateMetadata` can not be read or modified in the browser. `publicMetadata` can be read by the browser and can only be updated server-side or in the Clerk Dashboard, making it the safest and best choice for this use case.
+  There are three types of metadata:
 
-To build a basic RBAC system, first, you need to make `publicMetadata` available to the application directly from the session token. With `publicMetadata` attached directly to the user's session, a fetch and network request isn't required every time you need to access the data.
+  - `unsafeMetadata` - can be both read and modified in the browser, meaning users can modify this data. Therefore, you should validate `unsafeMetadata` before trusting it for sensitive operations.
+  - `privateMetadata` - cannot be accessed nor modified in the browser.
+  - `publicMetadata` - can be accessed in the browser, but only be modified server-side or in the Clerk Dashboard.
 
-In the Clerk Dashboard, navigate to [**Sessions**](https://dashboard.clerk.com/last-active?path=sessions). In the **Customize session token** section, select **Edit**. In the modal that opens, enter the following JSON. If you have already customized your session token, you may need to merge this with what you currently have.
+  Since `publicMetadata` can only be read but not modified in the browser, it is the safest and most appropriate choice for storing information
 
-```json
-{
-  "metadata": "{{user.public_metadata}}"
-}
-```
+  To build a basic RBAC system, you first need to make `publicMetadata` available to the application directly from the session token. By attaching `publicMetadata` to the user's session, you can access the data without needing to make a fetch or network request each time.
```

Comment:
> fetch or network request
This isn't really an 'or'. I'd say 'fetch and network request' where fetch describes the function/action and network request refers to the latency. Barring that I would say to use just fetch or just network request. eg

> without needing to make a fetch each time.

Reply:
[6bc6229](https://github.com/clerk/clerk-docs/pull/1560/commits/6bc62299498e9c94b7d9a789f9387098108452c6)